crates/walrus-service/src/node/storage.rs

```diff
@@ -634,9 +642,13 @@ impl Storage {
 
     /// Deletes the metadata and slivers for the provided [`BlobId`] from the storage.
     #[tracing::instrument(skip_all)]
-    pub async fn delete_blob_data(&self, blob_id: &BlobId) -> Result<(), TypedStoreError> {
+    pub async fn delete_blob_data(
+        &self,
+        blob_id: &BlobId,
+        update_blob_info: bool,
```

Comment:
Do we need this boolean flag? It seems like the only caller for this function is `process_blob_invalid_event`.